src/js/lib/monster.ui.js

```diff
@@ -2120,97 +2120,149 @@ define(function(require) {
-							color: '#18b309',
-							size: 100
-						},
-						{
-							key: 'good',
-							regex: new RegExp('^(?=.{8,})(((?=.*[A-Z])(?=.*[\\W_]))|((?=.*[A-Z])(?=.*[0-9]))|((?=.*[\\W_])(?=.*[0-9]))).*$'),
-							color: '#33db24',
-							size: 70
-						},
-						{
-							key: 'medium',
-							regex: new RegExp('^(?=.{6,})((?=.*[\\W_])|(?=.*[A-Z])|(?=.*[0-9])).*$'),
-							color: '#ffcc33',
-							size: 50
-						},
-						{
-							key: 'weak',
-							regex: new RegExp('^(?=.{6,}).*$'),
-							color: '#ff6a57',
-							size: 40
-						},
-						{
-							key: 'bad',
-							regex: new RegExp('^.+$'),
-							color: '#ff3d24',
-							size: 20
-						},
-						{
-							key: 'empty',
-							regex: new RegExp('^\\s*$'),
-							color: '#c0c0c9',
-							size: 0
-						}
-					],
-					strengthDisplay;
-
-				switch (display) {
-					case 'icon': {
-						strengthDisplay = $('<i class="fa fa-lock icon-small monster-password-strength" data-original-title="' + i18n.passwordStrength.empty + '" data-placement="' + tooltipPosition + '" data-toggle="tooltip"></i>');
-						input.on('keyup keypress change', function(e) {
-							$.each(regexes, function(key, val) {
-								if (val.regex.test(input.val())) {
-									strengthDisplay
-										.css('color', val.color)
-										.attr('data-original-title', i18n.passwordStrength[val.key])
-										.tooltip('fixTitle');
-									return false;
-								}
-							});
-						});
-						break;
-					}
-					case 'bar':
-					default: {
-						strengthDisplay = $('<div class="monster-password-strength"><div><span>' + i18n.passwordStrength.empty + '</span></div></div>');
-						input.on('keyup keypress change', function(e) {
-							$.each(regexes, function(key, val) {
-								if (val.regex.test(input.val())) {
-									strengthDisplay
-										.children('div')
-										.css({
-											backgroundColor: val.color,
-											width: val.size + '%'
-										}).children('span')
-										.html(i18n.passwordStrength[val.key]);
-									return false;
-								}
-							});
+		/**
+		 * Adds a password strength indicator for a specific input, in the form of either a bar, an emoji or a color-changing lock icon.
+		 * @param  {jQuery} input  Input on which the method will be applied. Best suited for an input of the `password` type.
+		 * @param  {Object} [options]  Indicator options
+		 * @param  {jQuery} [options.container]  Where to append the password strength display (by default, it will be appended after the `input`).
+		 * @param  {('bar'|'emoji'|'icon')} [options.display='bar']  Type of indicator to display: a bar, an emoji or a color-changing lock icon.
+		 * @param  {('top'|'bottom'|'left'|'right')} [options.tooltipPosition='top']  When the display is set to 'icon', you can choose the position of the tooltip on the icon.
+		 */
+		showPasswordStrength: function(input, pOptions) {
+			if (!input) {
+				throw new Error('You must provide at least one input field');
```

Comment:
It doesn't look like we can provide more than one input field though?

Reply:
Yes, that's true, only one input at a time is supported.